requirements.txt

```diff
@@ -6,5 +6,5 @@ future>=0.17.1  # required for builtins in setup.py
 # pyyaml>=3.13
 PyYAML>=5.1  # OmegaConf requirement >=5.1
 tqdm>=4.41.0
-fsspec>=0.8.0
+fsspec[http]>=0.8.1
```

Comment:
is this needed for this fix? we should probably let the user choose all extensions

Reply:
Yes, it is needed because PyTorch-Lightning (PL) uses http protocol & fsspec.  

PL's [`get_filesystem`][get_fs] requires `http` extras for http protocol handling (if no extras, it throw error.)  
At the same time, PL has a feature "restore from http" (example: test case [`test_strict_model_load_more_params`][restore_http].)  
But requrements.txt do NOT have `http` extras for `ffspec`.  

It means that **PL has been had hidden bug related to ffspec & http**, but it has not been came to the surface.  
This PR use `get_filesystem` in `restore`, then reveal the embedded bug (w/o http extras, CI test failed.)  

With `fsspec[http]>=0.8.1`, we can resolve this error.  
Alternatively PL could deprecate http acceptance feature by default (delete all http test case), but I prefer the former approach.  

[get_fs]:https://github.com/PyTorchLightning/pytorch-lightning/blob/91c64e9c82f118a9fa60012df89aa2dd217a7953/pytorch_lightning/utilities/cloud_io.py#L35
[restore_http]:https://github.com/PyTorchLightning/pytorch-lightning/blob/91c64e9c82f118a9fa60012df89aa2dd217a7953/tests/models/test_restore.py#L404